### PR TITLE
Move DoubleBuffering tests to a separate source file

### PR DESCRIPTION
**Pull request by drzejan2**
The changes are a part of preparations for Hopper support for double buffering implemented with TMA descriptors.

The scope of changes is to move double buffering tests from `test_gpu3.cpp` to separate file, `test_double_buffering.cpp`.

### CODE REVIEW

**Comment by drzejan2** [comment]
!build

**Review by naoyam — APPROVED** [review]
LGTM

File: test/test_double_buffering.cpp [hunk 1]
```diff
@@ -0,0 +1,686 @@
...
+#include <test/utils.h>
+#include <test/validator.h>
+
+namespace nvfuser {
+
+namespace {
+class DoubleBufferingTest : public NVFuserTest {};
+} // anonymous namespace
+
+TEST_F(DoubleBufferingTest, FusionDoubleBuffering1) {
```
Comment (zasdfgbnm) [inline test/test_double_buffering.cpp:19]: nit: remove the `Fusion` prefix in test name

**Review by zasdfgbnm — APPROVED** [review]

File: test/test_double_buffering.cpp (hunk 1 above)
Comment (drzejan2) [reply test/test_double_buffering.cpp:19]: Sure, I will update the PR and then integrate it.

Thanks!

**Pull request merged**